### PR TITLE
[core] silence warning for deprecated fn in Compression.cxx

### PR DESCRIPTION
**Pull request by silverweed**

### CODE REVIEW

File: core/zip/src/Compression.cxx [hunk 1]
```diff
@@ -25,24 +24,27 @@ namespace ROOT {
...
-    if (algorithm >= ROOT::ECompressionAlgorithm::kUndefinedCompressionAlgorithm) algo = 0;
+    if (algo >= ROOT::RCompressionSetting::EAlgorithm::kUndefined) algo = 0;
     return algo * 100 + compressionLevel;
   }
+#pragma GCC diagnostic pop
 
   std::string RCompressionSetting::AlgorithmToString(RCompressionSetting::EAlgorithm::EValues algorithm)
   {
      switch (algorithm) {
-     case EAlgorithm::EValues::kZLIB: return "zlib"; break;
```
Comment (vepadulano) [inline core/zip/src/Compression.cxx:41]: Why are the `break`s removed here?

File: core/zip/src/Compression.cxx [hunk 2]
```diff
@@ -25,24 +24,27 @@ namespace ROOT {
     return algo * 100 + compressionLevel;
   }
 
-  int CompressionSettings(ROOT::ECompressionAlgorithm algorithm,
-                          int compressionLevel)
+// Avoid spitting warnings for the use of ECompressionAlgorithm, as this function is itself deprecated
+#pragma GCC diagnostic push
```
Comment (vepadulano) [inline core/zip/src/Compression.cxx:28]: So this is a problem only for GCC? We should also check MSVC.

Comment (silverweed) [reply core/zip/src/Compression.cxx:28]: Yes, I'll check the windows tests when they are done.

File: core/zip/src/Compression.cxx (hunk 1 above)
Comment (silverweed) [reply core/zip/src/Compression.cxx:41]: They're useless after a `return`

**Review by vepadulano — APPROVED** [review]
Thanks a lot!

**Pull request merged**